src/dag.rs

```diff
@@ -368,23 +338,17 @@ mod tests {
     // changes, than have a clever (a.k.a buggy) test that tries to really just
     // assert what mattters.
     #[test_case(nodes([vec![1], vec![2, 3], vec![], vec![]]),
-                0, vec![0, 1, 3, 2]; "tree")]
+                0, vec![0, 1, 2, 3]; "tree")]
     #[test_case(nodes([vec![1], vec![2, 3], vec![], vec![]]),
-                1, vec![1, 3, 2]; "tree non root")]
+                1, vec![1, 2, 3]; "tree non root")]
     #[test_case(nodes([vec![1], vec![2, 3], vec![], vec![],
                        vec![5], vec![6, 7], vec![], vec![]]),
-                0, vec![0, 1, 3, 2]; "trees 1")]
+                0, vec![0, 1, 2, 3]; "trees 1")]
     #[test_case(nodes([vec![1], vec![2, 3], vec![], vec![],
                        vec![5], vec![6, 7], vec![], vec![]]),
-                4, vec![4, 5, 7, 6]; "trees 2")]
-    // BUG: This test fails:
-    //      assertion `left == right` failed
-    //      left: [0, 2, 3, 4, 1, 4]
-    //      right: [0, 2, 3, 1, 4]
-    // TODO: Uncomment once bug is fixed.
```

Comment:
Lol this is what happens when you work without code review... I must have written a TODO in here, forgotten about it, come back to this like 2 weeks later, seen that all the tests pass, and gone "cool, merge it"

Reply:
No, you added these tests! That's a relief. I was confused because Github shows commits in the wrong order (long standing issue which for some reason they refuse to fix)